components/server/src/ome/logic/AdminImpl.java

```diff
@@ -1442,4 +1557,29 @@ private void adminOrPiOfNonUserGroups(ExperimenterGroup defaultGroup,
                 nonUserGroupGroups.toArray(new ExperimenterGroup[0]));
     }
 
+    /**
+     * @param privileges the privileges that are expected to be among those of the {@link AdminPrivilege} enumeration values
+     * @throws ApiUsageException if passed an unknown privilege
+     */
+    private void assertKnownPrivileges(Iterable<AdminPrivilege> privileges) {
+        for (final AdminPrivilege privilege : privileges) {
+            if (adminPrivileges.getPrivilege(privilege.getValue()) == null) {
+                throw new ApiUsageException("unknown light administrator privilege: " + privilege.getValue());
+            }
+        }
+    }
+
+    /**
+     * @return the light administrator privileges that apply to the current session
+     */
+    private Set<AdminPrivilege> getCurrentAdminPrivilegesForSession() {
+        if (isAdmin()) {
+            final String hql = "FROM Session s LEFT OUTER JOIN FETCH s.sudoer WHERE s.id = :id";
+            final Parameters params = new Parameters().addId(getEventContext().getCurrentSessionId());
+            final ome.model.meta.Session session = iQuery.findByQuery(hql, params);
```

Comment:
Not that I've noticed but I haven't much checked (can changes in CI timings help?) and the internal security infrastructure, which I figure is where the frequent checks occur, is making use of an explicit cache of them, and where they're not in an OMERO cache I'd guess they're probably soon in a Hibernate one. Could certainly look at preloading sudoer somehow? Maybe even preloading privileges depending on how session cache stuff would handle changes in privileges during long-lived session.